### PR TITLE
Bumped `mesos-overlay-modules`

### PR DESCRIPTION
**Pull request by asridharan**
* Added unit-test cases for `mesos-overlay-modules`. We can run `make
check`.
* Re-factored Agent and Master modules for the unit-test.
* Fixed a discrepancy in the information exposed in the
overlay-master/state` endpoint and the `overlay-agent/overlay` endpoint.
* Ensuring that `DOCKER-ISOLATION` bypass iptable rules are installed
only if the `docker-bridge` is enabled in the Agent module.

### CODE REVIEW

**Comment by asridharan** [comment]
Ready to merge !!

File: packages/mesos-overlay-modules/buildinfo.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
     "docker": "dcos-builder",
     "single_source" : {
       "kind": "git",
-      "git": "https://github.com/dcos/mesos-overlay-modules.git",
-      "ref": "9ea9dab6cbb117ce244524bfb6d119e96b5289a8",
+      "git": "git@github.com:dcos/mesos-overlay-modules.git",
```
Comment (cmaloney) [inline packages/mesos-overlay-modules/buildinfo.json]: Need to keep using https.

Comment (asridharan) [inline packages/mesos-overlay-modules/buildinfo.json]: My bad will fix !!

**Comment by asridharan** [comment]
Update the github reference.

**Comment by cmaloney** [comment]
🚢

**Pull request merged**